modules/javafx.controls/src/main/java/javafx/scene/control/skin/PaginationSkin.java

```diff
@@ -194,6 +194,8 @@ public PaginationSkin(final Pagination control) {
         nextStackPane.getStyleClass().add("page");
         nextStackPane.setVisible(false);
 
+        resetIndexes(true);
```

Comment:
This will end up calling into the control in a couple places, which the PR description says you are trying to avoid.

Reply:
true, but since it sets to the same value it will be a no-op

Reply:
Right.